src/parse_file.py

```diff
@@ -458,7 +460,11 @@ def process_label(label: str, *, glabel: bool) -> None:
                     lineno=lineno,
                     synthetic=False,
                 )
-                instr: Instruction = parse_instruction(line, meta, arch)
+                instr: Instruction
+                if mips_file.current_function is not None:
+                    instr = parse_instruction(line, meta, arch, mips_file.current_function.used_reg_names)
+                else:
+                    instr = parse_instruction(line, meta, arch, UsedRegNames())
```

Comment:
very small nit: I think I'd prefer something like this, to minimize the code in the branch
```suggestion
                if mips_file.current_function is not None:
                    used_reg_names = mips_file.current_function.used_reg_names
                else:
                    used_reg_names = UsedRegNames()
                instr = parse_instruction(line, meta, arch, used_reg_names)
```

Reply:
Simon said the same thing, ha